src/common/_modules/private_endpoint/locals.tf

```
@@ -0,0 +1,9 @@
+locals {
+  private_endpoints = {
+    "cgn-psql" = {
+      resource_id         = data.azurerm_postgresql_server.cgn_psql.id
+      subresource_names   = ["postgresqlServer"]
+      private_dns_zone_id = data.azurerm_private_dns_zone.cgn_psql_private_dns_zone.id
```

To make this work, it's important to add an entry like the following [here](https://github.com/pagopa/io-infra/blob/03408acd89d68f89c32c016ec6e4867414b0d3d6/src/common/_modules/global/modules/dns/outputs.tf#L34):
`postgres  = azurerm_private_dns_zone.privatelink_postgres_database_azure_com`


```suggestion
      private_dns_zone_id = module.global.dns.private_dns_zones.postgres.id
```

---

Since it’s a `map` where you can specify as many `pep` as you need, I’ve passed the entire `private_dns_zones` output so you can choose one from it.

---

Agree! It's very important to add the output in the dns module as suggested in the corpus of the comment